File: src-lang/xt/mcp/node/proxy_base.clj
(ns xt.mcp.node.proxy-base
  (:require [hara.lang :as l]))

(l/script :xtalk
  {:require [[xt.lang.spec-base :as xt]
             [xt.substrate :as substrate]
             [xt.mcp.node.proxy-util :as proxy-util]]})

(def.xt MESSAGE_ACTION "@xt.mcp/message")

(defn.xt request-proxy
  "forwards an MCP message to the server node"
  {:added "4.1"}
  [space args request node]
  (return (proxy-util/request-proxy space args request node)))

(defn.xt init-proxy-handlers
  "installs client-side handlers mirroring the MCP server action ids"
  {:added "4.1"}
  [node]
  (substrate/register-handler node -/MESSAGE_ACTION -/request-proxy nil)
  (return node))

File: packages-gen/js/libs/xt-mcp/node/proxy-base.js
const proxy_util = require("@xtalk/mcp/node/proxy-util.js")

const substrate = require("@xtalk/substrate/substrate.js")

var MESSAGE_ACTION = "@xt.mcp/message";

function request_proxy(space,args,request,node){
  return proxy_util.request_proxy(space,args,request,node);
}

function init_proxy_handlers(node){
  substrate.register_handler(node,MESSAGE_ACTION,request_proxy,null);
  return node;
}

module.exports = {
  ["MESSAGE_ACTION"]:MESSAGE_ACTION,
  ["request_proxy"]:request_proxy,
  ["init_proxy_handlers"]:init_proxy_handlers
}